File: backends/go-gin/internal/domain/client/condition.go
package client

import "time"

// Condition はクライアント検索条件です。
type Condition struct {
	Keyword   *string
	StartFrom *time.Time
	StartTo   *time.Time
	Statuses  []int
	Offset    int
	Limit     int
	Sort      string
	SortType  string
}

File: backends/kotlin-ktor/src/main/kotlin/com/authorization/domain/client/Condition.kt
/*
 * クライアントドメイン検索条件モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.domain.client

import java.time.LocalDateTime

/**
 * クライアント検索条件です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class Condition(
    val keyword:   String?          = null,
    val startFrom: LocalDateTime?   = null,
    val startTo:   LocalDateTime?   = null,
    val statuses:  List<Int>        = emptyList(),
    val offset:    Int              = 0,
    val limit:     Int              = 10,
    val sort:      String?          = null,
    val sortType:  String?          = null,
)
